menu.py

```diff
@@ -1,38 +1,452 @@
+    # kVK_Mute: '',
+    kVK_F18: "f18",
+    kVK_F19: "f19",
+    kVK_F20: "f20",
+    kVK_F5: "f5",
+    kVK_F6: "f6",
+    kVK_F7: "f7",
+    kVK_F3: "f3",
+    kVK_F8: "f8",
+    kVK_F9: "f9",
+    kVK_F11: "f11",
+    kVK_F13: "f13",
+    kVK_F16: "f16",
+    kVK_F14: "f14",
+    kVK_F10: "f10",
+    kVK_F12: "f12",
+    kVK_F15: "f15",
+    # kVK_Help: '',
+    kVK_Home: "home",
+    kVK_PageUp: "pageup",
+    kVK_ForwardDelete: "delete",
+    kVK_F4: "f4",
+    kVK_End: "end",
+    kVK_F2: "f2",
+    kVK_PageDown: "pagedown",
+    kVK_F1: "f1",
+    kVK_LeftArrow: "left",
+    kVK_RightArrow: "right",
+    kVK_DownArrow: "down",
+    kVK_UpArrow: "up",
+    # kVK_ISO_Section: '',
+    # kVK_JIS_Yen: '',
+    # kVK_JIS_Underscore: '',
+    # kVK_JIS_KeypadComma: '',
+    # kVK_JIS_Eisu: '',
+    # kVK_JIS_Kana: '',
+}
+
+
+def active_menu_bar():
     return ui.active_app().children.find_one(AXRole="AXMenuBar", max_depth=0)
 
 
+def selected_menu_and_path():
+    """Returns selected element in menu bar, and path to it"""
+    selected_menu = active_menu_bar()
+    menu_path = []
+    while True:
+        if not selected_menu.AXSelectedChildren:
+            break
+        selected_menu = selected_menu.AXSelectedChildren[0]
+
+        if not (title := selected_menu.get("AXTitle")):
+            return selected_menu, menu_path
+
+        menu_path.append(selected_menu.AXTitle)
+        # XXX(nriley) this is True even if list empty - report Talon bug? children also doesn't support len()
+        # print(selected_menu.children, bool(selected_menu.children))
+        try:
+            selected_menu = selected_menu.children.find_one(
+                AXRole="AXMenu", max_depth=0
+            )
+        except ui.UIErr:
+            break
+
+    return selected_menu, menu_path
+
+
+def mouse_pos_menu_and_path():
+    """Returns element in menu bar under cursor, and path to it"""
+    menu_path = []
+    element_at_mouse_pos = ui.element_at(*ctrl.mouse_pos())
+    if element_at_mouse_pos.AXRole in ("AXMenu", "AXMenuItem", "AXMenuBarItem"):
+        menu = element_at_mouse_pos
+        while True:
+            if menu.AXRole in ("AXMenuItem", "AXMenuBarItem"):
+                menu_path.append(menu.AXTitle)
+            menu = menu.AXParent
+            if menu.AXRole == "AXMenu":
+                menu = menu.AXParent
+            elif menu.AXRole == "AXMenuBar":
+                menu_path.reverse()
+                break
+        else:  # not in menu bar
+            return None, []
+
+    return element_at_mouse_pos, menu_path
+
+
+def selected_menu_path_strategy():
+    """Returns 'selected' element in menu bar, path to it and strategy used to find it"""
+    selected_menu, selected_menu_path = selected_menu_and_path()
+
+    if (
+        selected_menu_path
+        and hasattr(ui, "element_at")
+        and selected_menu.AXRole != "AXMenuItem"
+    ):
+        element_at_mouse_pos, mouse_pos_menu_path = mouse_pos_menu_and_path()
```

Comment:
```suggestion
        element, menu_path = mouse_pos_menu_and_path()
```

Prefer shorter variable names if the function scope is small enough that they won't get crowded.

Reply:
Naming has been painful here and I am sure not happy with the existing names, but I'm not sure if this improves readability. Was trying to distinguish them from `selected_menu` and `selected_menu_path` respectively.

Reply:
Yeah, I don't feel strongly. Fair enough.